[Feature] Firebase Auth with Google Sign-In

tranhaidangdz opened the pull request:
xây dựng tầng data gồm : data source and repository

tranhaidangdz committed:
[Setup] Firebase configuration:
- Add google-services.json (Android) / GoogleService-Info.plist (iOS)
- Add firebase dependencies and google-services plugin

tranhaidangdz committed:
[Feat] Firebase Auth with Google Sign-In:Add data source and repository

tranhaidangdz committed:
[Feat] Firebase Auth with Google
- hoàn thiện  giao diện login screen
- kết nối splash->welcome->login->onBoarding bằng navRoute

DinhNgocThe commented on app/src/main/java/com/utc/driverxy/data/domain/usecase/SignInWithGoogleUseCase.kt:
Domain là tầng Business logic, không nằm trong data. Business logic chỉ chứa code logic, không lieen quan đến android framework context, lifecycle...

DinhNgocThe commented on app/src/main/java/com/utc/driverxy/data/repository/AuthRepository.kt:
Interface này là tầng domain business logic, nằm trong package domain -> repository.
Đổi tên thành UserRepository để gom nhóm các hành động signInWithGoogle, getUserInforFromFireStore...
Tương tự đổi tên với impl, impl dã đúng package rồi

DinhNgocThe commented on app/src/main/java/com/utc/driverxy/di/KoinApp.kt:
Nhiều dòng thừa vậy, nguyên tắc kế thừa KoinApp : Application(), kiểu dữ liệu trả về thì val a: String k có space trước :

tranhaidangdz committed:
[Feat] Firebase Auth with Google
- hoàn thiện  giao diện login screen
- kết nối splash->welcome->login->onBoarding bằng navRoute

tranhaidangdz committed:
[Fix] đổi tên repo -> userRepository

tranhaidangdz committed:
[Fix] fix lại khoảng cách giữa các hàm

tranhaidangdz committed:
[Fix] fix lại khoảng cách giữa các hàm

Temp-art05 commented on app/src/main/java/com/utc/driverxy/data/repository/UserRepositoryImpl.kt:
```diff
@@ -0,0 +1,16 @@
+import com.utc.driverxy.domain.repository.AuthRepository
+
+class AuthRepositoryImpl(
+    private val remote: GoogleAuthDataSource
```
nhứng cái khai báo này thì viết tường minh private val googleAuthDataSource: GoogleAuthDataSource. Áp dụng mọi nơi

tranhaidangdz committed:
[Fix] đổi lại tên AuthRepo->UserRepo

vinhtruong204 committed:
setup login with google
